kani-compiler/src/kani_middle/attributes.rs

```diff
@@ -83,7 +85,10 @@ pub fn is_test_harness_closure(tcx: TyCtxt, def_id: DefId) -> bool {
 }
 
 /// Extract all Kani attributes for a given `def_id` if any exists.
+/// We only extract attributes for harnesses that are local to the current crate.
 pub fn extract_harness_attributes(tcx: TyCtxt, def_id: DefId) -> Option<HarnessAttributes> {
+    // Abort if not local.
+    def_id.as_local()?;
```

Comment:
Is this to avoid attribute extraction from external code?

Reply:
Yes. So we don't collect a harness from a dependency.